pepper-apis/src/main/java/org/broadinstitute/ddp/util/EmailBlasterCLI.java

```diff
@@ -0,0 +1,141 @@
+        options.addOption("g", "guids-file", true, "path to file of user guids");
+        options.addOption("e", "sender-email", true, "sender email");
+        options.addOption("f", "sender-name", true, "name of sender");
+        options.addOption("s", "study", true, "study guid");
+        options.addOption("t", "template-id", true, "sendgrid template id");
+
+        CommandLineParser parser = new DefaultParser();
+        CommandLine cmd = parser.parse(options, args);
+
+        HelpFormatter formatter = new HelpFormatter();
+        if (cmd.hasOption("help")) {
+            formatter.printHelp(DISPLAY_WIDTH, USAGE, "", options, "");
+            return;
+        }
+
+        Config cfg = ConfigFactory.load();
+        String sendgridApiKey = cfg.getString(ConfigFile.SENDGRID_API_KEY);
+        TransactionWrapper.init(
+                new TransactionWrapper.DbConfiguration(TransactionWrapper.DB.APIS, 1,
+                        cfg.getString(ConfigFile.DB_URL)));
+
+        String studyGuid = cmd.getOptionValue("s");
+        String fromName = cmd.getOptionValue("f");
+        String fromEmail = cmd.getOptionValue("e");
+        String templateId = cmd.getOptionValue("t");
+        File guidsFile = new File(cmd.getOptionValue("g"));
+
+        List<String> guids = IOUtils.readLines(new FileReader(guidsFile));
+
+        new EmailBlasterCLI(sendgridApiKey).sendEmail(fromName, fromEmail, templateId, studyGuid, guids);
+
+        System.exit(0);
+    }
+
+    public void sendEmail(String fromName, String fromEmail, String sendgridTemplateId, String studyGuid, Collection<String> recipientGuids) {
+
+        final Set<String> auth0UserIds = new TreeSet<>();
+        final Map<String, Map<String, String>> personalizationByAuth0Id = new HashMap<>();
+
+        TransactionWrapper.useTxn(handle -> {
+            JdbiUmbrellaStudy jdbiStudy = handle.attach(JdbiUmbrellaStudy.class);
+            JdbiAuth0Tenant tenantDao = handle.attach(JdbiAuth0Tenant.class);
+            JdbiUser userDao = handle.attach(JdbiUser.class);
+            StudyDto studyDto = jdbiStudy.findByStudyGuid(studyGuid);
+
+            Auth0TenantDto tenantDto = tenantDao.findByStudyGuid(studyDto.getGuid());
+            Auth0ManagementClient mgmtClient = Auth0Util.getManagementClientForDomain(handle, tenantDto.getDomain());
+            Auth0Util auth0Util = new Auth0Util(tenantDto.getDomain());
+
+            for (String recipientGuid : recipientGuids) {
+                UserDto userDto = userDao.findByUserGuid(recipientGuid);
+                UserProfile userProfile = handle.attach(UserProfileDao.class).findProfileByUserGuid(userDto.getUserGuid()).get();
+                // todo add other template vars
+                personalizationByAuth0Id.put(userDto.getAuth0UserId(), new HashMap<>());
+                personalizationByAuth0Id.get(userDto.getAuth0UserId()).put(DDP_PARTICIPANT_FIRST_NAME, userProfile.getFirstName());
```

Comment:
I believe substitution works for only Legacy SendGrid templates. 
Wee might want to mention that limitation in the code so that someone will not create a dynamic template